CONTRIBUTING.md

```diff
@@ -0,0 +1,111 @@
+   of the additions/modifications.
+3. Make your changes following the code style used in the samples. If you
+   are adding a new sample or library, follow the steps described in the
+   [Contributing a new sample](#contributing-a-new-sample) and
+   [Contributing a new library](#contributing-a-new-library) guides.
+4. Execute your code and verify the new functionality or fix works properly.
+5. Submit a pull request ([how to create a pull request][pull-request]).
+
+A project developer will review your work and then merge your request into the
+project, or come back to you with comments and/or questions.
+
+### Contributing a new sample
+
+XBee MicroPython samples are located in the `samples/` directory of the
+repository's root path. Samples are organized in categories, which are 
+directories that group samples by functionality.
+
+You can create and nest categories as needed, but before doing so, make sure
+there is not already a category that meets your needs.
+
+#### Sample contents
+
+* A sample requires a directory with the sample ID as name. This ID must be a
+  short name describing the example purpose, in lowercase and without blank
+  spaces. 
+* Each sample must include at least the following files inside its directory:
+  * `README.md` This file describes the sample functionality, requirements,
+    setup, etc. and is used by the **XBee MicroPython plug-in for PyCharm**
+    to list the sample in the import samples wizard. For that reason, it must
+    have a specific structure with required sections. You can copy and modify
+    file [README_TEMPLATE.md](samples/README_TEMPLATE.md) from `samples/`
+    directory for reference.
+  * `main.py` This file contains the source code of the sample and its name
+    cannot be changed. See other existing examples and follow the same code
+    style to write yours.
+* Other directories or files (source code or not) can be added to the sample
+  as needed.
+    
+### Contributing a new library
+
+XBee MicroPython libraries are located in the `lib/` directory of the
+repository's root path. libraries are organized in categories, which are 
+directories that group them by functionality.
+
+You can create and nest categories as needed, but before doing so, make sure
+there is not already a category that meets your needs.
+
+#### Library contents
+
+* A library requires a directory with the library ID as name. The name of the
+  directory must be a short name describing the example purpose, in lowercase
+  and without blank spaces.
+* Each library must include at least the following files inside its directory:
+  * `README.md` This file describes the library functionality and requirements
+    and is used by the **XBee MicroPython plug-in for PyCharm** to list the
+    library when creating a new project. For that reason, it must have a
+    specific structure with required sections. You can copy and modify file
+    [README_TEMPLATE.md](lib/README_TEMPLATE.md) from `lib/` directory for
```

Comment:
I think it makes sense to have a note here, and maybe within the sample files, that the lines that start with `>` (block quotes) should be removed when contributing. Otherwise people might contribute libraries with description "Brief description of the library. Example: ..."

Reply:
Done.